src/Domain/DonationRepository.php

```diff
@@ -662,4 +666,48 @@ public function removeAllFundingWithdrawalsForDonation(Donation $donation): void
             }
         });
     }
+
+    private function setSalesforceIdIfNeeded(Donation $donation, string $salesforceId): void
```

Comment:
Worth using the Salesforce18Id VO that we have here and in the related functions? 

```suggestion
    private function setSalesforceIdIfNeeded(Donation $donation, string $salesforceId): void
```

Reply:
Done in pairing